src/examples/ExamplesPanel.ts

```diff
@@ -82,41 +82,48 @@ export class ExamplesPlanel {
 
     this.panel.onDidDispose(() => this.dispose(), null, this.disposables);
 
-    this.panel.webview.onDidReceiveMessage((message) => {
+    this.panel.webview.onDidReceiveMessage(async (message) => {
       switch (message.command) {
         case "openExampleProject":
           if (message.project_path && message.name) {
-            vscode.window
-              .showOpenDialog({
-                canSelectFolders: true,
-                canSelectFiles: false,
-                canSelectMany: false,
-              })
-              .then((selectedFolder: vscode.Uri[]) => {
-                if (selectedFolder && selectedFolder[0].fsPath) {
-                  const resultFolder = path.join(
-                    selectedFolder[0].fsPath,
-                    message.name
-                  );
-                  ensureDir(resultFolder)
-                    .then(() => {
-                      utils.copyFromSrcProject(
-                        message.project_path,
-                        resultFolder
-                      );
-                      const projectPath = vscode.Uri.file(resultFolder);
-                      vscode.commands.executeCommand(
-                        "vscode.openFolder",
-                        projectPath
-                      );
-                    })
-                    .catch((err) => {
-                      Logger.errorNotify("Error copying ESP-IDF example", err);
-                    });
-                } else {
-                  vscode.window.showInformationMessage("No folder selected");
-                }
-              });
+            const selectedFolder = await vscode.window.showOpenDialog({
+              canSelectFolders: true,
+              canSelectFiles: false,
+              canSelectMany: false,
+            });
+            if (!selectedFolder) {
+              vscode.window.showInformationMessage("No folder selected");
+              return;
+            }
+            try {
```

Comment:
One suggestion here, going forward we have to show very friendly and concise message to the user with an option to open documentation, so that they can see why this error is happening and how they can unblock themselves

Something similar [I tried here](https://github.com/espressif/vscode-esp-idf-extension/pull/170/files#diff-04bba6a35cad1c794cbbe677678a51de13441b7a6ee8592b7b50be1f05c6f626R1401), maybe later we can have a task to extract this into a function like `showErrorMessageWithDocumentation(msg: string, error: Error, documentationLink: URL, meta: any)`

Reply:
I understand the approach but what would the documentation to show here ? We would need to list a document with common errors and how to fix them (here for example would be like file permissions or toolchain not found ?)

Reply:
> need to list a document with common errors and how to fix them

Not really sure, but for now what we can do is since we already have the readme in `/docs` we can add a section of _Troubleshooting_ for all the docs there.

And then link any error in a feature to the specific feature docs.

What do you think?